### PR TITLE
work in progress - adding dashboard backend data pipeline

### PR DESCRIPTION
**Pull request by mjperrone**
refactoring needed, and still need to:

1. write code to load csvs into the db
2. write script to generate csv representing one of the data views
3. add the ETL script to a cron job
4. add the generate csv to a cron job
5. expose the generated csv in nginx

### CODE REVIEW

File: dashboard/generate_stats.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+        r = requests.post(
+            "https://graph.facebook.com/v2.5",
+            data={"batch": batch_req},
+            headers={"Authorization": "Bearer {}".format(os.environ["FACEBOOK_APP_ACCESS_TOKEN"])}
+        )
+        data = r.json()
+        for res in data:
+            community = json.loads(res['body'])['id']
+            if community[0] == 'h':
```
Comment (samertm) [inline dashboard/generate_stats.py]: are FB ids [0-9] numbers? maybe community.startswith('http') would be more clear.

File: dashboard/generate_stats.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+def links_to_ids(links):
+    """Uses the Facebook Graph API to get the group/page's id from their link"""
+
+    # This uses's the batch request feature
+    # (https://developers.facebook.com/docs/graph-api/making-multiple-requests)
+    # to make many logical requests within one http request, saving the overhead
+    # cost
+
+    ids = []
+    links = links[:49]
```
Comment (samertm) [inline dashboard/generate_stats.py]: you limit the links to 50 here, but you chunk them in groups of 50 below?

**Comment by mjperrone** [comment]
lol @samertm don't CR it. You told me to put it up as it is so you could see how it works. It's very not done, there's a lot that doesn't make sense and won't stay.

File: dashboard/generate_stats.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    """Grab the facebook links for each local chapter from airtable."""
+    raw_data = get_all_records("Chapters", "Main View")
+    chapter_links = []
+    for row in raw_data:
+        fields = row["fields"]
+        if "Facebook" in fields:
+            try:
+                chapter_links.append(fields["Facebook"])
+            except:
+                # TODO: log it, error it, email it, something like that
```
Comment (samertm) [inline dashboard/generate_stats.py]: throwing an error would make sense b/c I assume we'll get an email from the cron job that runs this? is that true?

Comment (samertm) [inline dashboard/generate_stats.py]: also, not sure how this would throw an error b/c you have the `if "Facebook" in fields` check above

**Comment by samertm** [comment]
@mjperrone hahaha kk, sorry for jumping the gun :P

**Comment by samertm** [comment]
overall it looks good tho :+1:

**Pull request closed without merging**